### DISCUSSION TITLE
Extracting Fyrox UI as a standalone project

### DISCUSSION THREAD

**Discussion by Umenokin** [FyroxEngine/Fyrox#347] [Ideas]
Hi @mrDIMAS 
I am not a Fyrox user. 
I noticed your Patreon account and such small support considering the amount of effort you put into development.
It will take an infinitely impossible amount of effort trying to provide production quality comparable to even Godot.
But what is rust community is definitely struggling with is a production-ready UI library for desktop-like applications. Especially if we are talking about retained mode library.

Maybe you should extract Fyrox UI as a stand-alone library without a game development accent?
1. You will be able to attract nongame industries
2. You will be able to promote it as competitively production-ready way faster than a whole engine.
3. It is part of your engine anyway, so it should not slow you down or move a lot of your focus away from the engine.
4. Many professional developers would be interested in UI even if it does not cover 100% of their needs but how many experienced game developers would choose the Fyrox engine in its current state as an alternative to Unreal Engine, Unity, or Godot besides educational purposes?
5. UI extraction is pure marketing to attract users from other industries.

You are a one-man army!
Best

**Comment by mrDIMAS** [comment]
Hi, `fyrox-ui` is a standalone library, it just needs native windowing environment with a renderer. It can be used to create UI for any applications. I think it is possible to use `fyrox-ui` to build some sort of a UI framework on top of it using `winit` + some renderer like OpenGL.

**Reply by Umenokin** [reply]
This is exactly what I am pointing out. It is already a separate crate but needs to be a separate product. With its own, non-game examples, release announcements, changelog, etc.
